src/consume.ts

```diff
@@ -656,18 +658,17 @@ function messageViewerStartPollingCommand(
           serialized.value.includes(index),
         );
 
-        // i want to drop the comment in favor of filename and possibly do a preview tab
-        workspace
-          .openTextDocument({
-            content: `// message ${message.key} from ${topic.name}\n${JSON.stringify(payload, null, 2)}`,
-            language: "jsonc",
+        // read-only document buffer with the message content
+        const filename = `${topic.name}-message-${index}.json`;
+        const uri: Uri = new MessageDocumentProvider().resourceToUri(payload, filename);
+        window
+          .showTextDocument(uri, {
+            preview: true,
+            viewColumn: ViewColumn.Beside,
+            preserveFocus: false,
           })
-          .then((preview) => {
-            return window.showTextDocument(preview, {
-              preview: false,
-              viewColumn: ViewColumn.Beside,
-              preserveFocus: false,
-            });
+          .then((editor) => {
+            languages.setTextDocumentLanguage(editor.document, "jsonc");
```

Comment:
I guess it can be just `json` since it's read-only and we don't append any comments.

Reply:
Ah true, I wasn't sure if we wanted to add comments back in  - if we do, we'll have to account for any `key`-as-JSON instances. Changing to `json` for now https://github.com/confluentinc/vscode/pull/799/commits/a9e794a45a328cc0f940986bd46afec35f682ba1